zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala

```diff
@@ -21,61 +23,179 @@ import scala.jdk.CollectionConverters._
  * Credits for this work go to Nick Pavlov (https://github.com/gurinderu), Guillaume Bécan (https://github.com/gbecan)
  * and the Conduktor (https://www.conduktor.io/) devs team.
  */
+//noinspection SimplifyUnlessInspection,SimplifyWhenInspection
 object SslHelper {
-  def validateEndpoint(bootstrapServers: List[String], props: Map[String, AnyRef]): Task[Unit] =
-    ZIO
-      .unless(
-        props
-          .get("security.protocol")
-          .exists {
-            case x: String if x.toUpperCase().contains("SSL") => true
-            case _                                            => false
+
+  /**
+   * A private exception that we use to "tag" some exceptions that we potentially want to ignore.
+   */
+  private final case class ConnectionError(cause: Throwable) extends NoStackTrace
+
+  // ⚠️ Must not do anything else than calling `doValidateEndpoint`. The algorithm of this function must be completely contained in `doValidateEndpoint`.
+  def validateEndpoint(bootstrapServers: List[String], props: Map[String, AnyRef]): IO[KafkaException, Unit] =
+    doValidateEndpoint(SocketChannel.open)(bootstrapServers, props)
+
+  /**
+   * We use this private function so that we can easily manipulate the `openSocket` function in unit-tests.
+   */
+  private[utils] def doValidateEndpoint(
+    unsafeOpenSocket: InetSocketAddress => SocketChannel // Handy for unit-tests
+  )(bootstrapServers: List[String], props: Map[String, AnyRef]): IO[KafkaException, Unit] = {
+    @inline def `request.timeout.ms`: Duration = {
+      val defaultValue = 30.seconds
+
+      props.get("request.timeout.ms") match {
+        case None => defaultValue
+        case Some(raw) =>
+          try {
+            val v = raw.toString.toLong
+            if (v <= 0) defaultValue else v.millis
```

Comment:
Often `0` is used to disable timeouts. Is that the case here as well (if you ask me: no)? But if its 'yes', the code should be changed to:

```suggestion
            if (v < 0) defaultValue else v.millis
```

Reply:
We can't use `0` here as it'd mean "infinite"